easybuild/easyblocks/b/bzip2.py

```diff
@@ -40,14 +42,32 @@ class EB_bzip2(ConfigureMake):
     def configure_step(self):
         """Set extra configure options (CC, CFLAGS)."""
 
+        self.cfg.update('prebuildopts', "make -f Makefile-libbz2_so && ")
```

Comment:
you also need to take `self.cfg['buildopts']` into account when running `make -f Makefile-libbz2_so`, otherwise the `.so` is built with `gcc` regardless of the toolchain being used:

```
== 2016-05-08 15:36:15,406 run.py:392 DEBUG cmd " make -f Makefile-libbz2_so &&   make -j 10  CC="icc"  CFLAGS='-Wall -Winline -fPIC -O2 -xHost -ftz -fp-speculation=safe -fp-model source -g $(BIGFILES)' " exited with exitcode 0 and output:
gcc -fpic -fPIC -Wall -Winline -O2 -g -D_FILE_OFFSET_BITS=64 -c blocksort.c
gcc -fpic -fPIC -Wall -Winline -O2 -g -D_FILE_OFFSET_BITS=64 -c huffman.c
gcc -fpic -fPIC -Wall -Winline -O2 -g -D_FILE_OFFSET_BITS=64 -c crctable.c
gcc -fpic -fPIC -Wall -Winline -O2 -g -D_FILE_OFFSET_BITS=64 -c randtable.c
gcc -fpic -fPIC -Wall -Winline -O2 -g -D_FILE_OFFSET_BITS=64 -c compress.c
gcc -fpic -fPIC -Wall -Winline -O2 -g -D_FILE_OFFSET_BITS=64 -c decompress.c
gcc -fpic -fPIC -Wall -Winline -O2 -g -D_FILE_OFFSET_BITS=64 -c bzlib.c
gcc -shared -Wl,-soname -Wl,libbz2.so.1.0 -o libbz2.so.1.0.6 blocksort.o huffman.o crctable.o randtable.o compress.o decompress.o bzlib.o
gcc -fpic -fPIC -Wall -Winline -O2 -g -D_FILE_OFFSET_BITS=64 -o bzip2-shared bzip2.c libbz2.so.1.0.6
bzip2.c: In function ‘testStream’:
bzip2.c:557:37: warning: variable ‘nread’ set but not used [-Wunused-but-set-variable]
    Int32   bzerr, bzerr_dummy, ret, nread, streamNo, i;
                                     ^
rm -f libbz2.so.1.0
ln -s libbz2.so.1.0.6 libbz2.so.1.0
rm -f libbz2.a
icc -Wall -Winline -fPIC -O2 -xHost -ftz -fp-speculation=safe -fp-model source -g -D_FILE_OFFSET_BITS=64 -c bzip2.c
ar cq libbz2.a blocksort.o huffman.o crctable.o randtable.o compress.o decompress.o bzlib.o
icc -Wall -Winline -fPIC -O2 -xHost -ftz -fp-speculation=safe -fp-model source -g -D_FILE_OFFSET_BITS=64 -c bzip2recover.c
ranlib libbz2.a
...
```